pysb/core.py

```diff
@@ -404,8 +406,28 @@ def is_site_concrete(self):
         Return a bool indicating whether the pattern is 'site-concrete'.
 
         'Site-concrete' means all sites have specified conditions."""
-        # assume __init__ did a thorough enough job of error checking that this is is all we need to do
-        return len(self.site_conditions) == len(self.monomer.sites)
+        if len(self.site_conditions) != len(self.monomer.sites):
+            return False
+        for site_name, site_val in self.site_conditions.items():
+            if isinstance(site_val, STRING_TYPES):
```

Comment:
Could you use `basestring` directly here instead of defining `STRING_TYPES`?

Reply:
Gah, indeed. Fixed.